pkg/api/config/config.go

```diff
@@ -17,8 +17,8 @@ var (
 	BinaryType string //nolint: gochecknoglobals
 	GoVersion  string //nolint: gochecknoglobals
 
-	openIDSupportedProviders = [...]string{"google", "gitlab", "dex"} //nolint: gochecknoglobals
-	oauth2SupportedProviders = [...]string{"github"}                  //nolint: gochecknoglobals
+	openIDSupportedProviders = [...]string{"google", "gitlab", "dex", "oidc"} //nolint: gochecknoglobals
```

Comment:
@peusebiu any reason we need to keep "dex" around. It could be just another "oidc" instance.

Reply:
The reason was to facilitate transition

Reply:
@babs we are in a rc window, so changes to be expected. Thanks for the thoughtful code though.

Reply:
@rchincha would you like me to remove dex ?
I'll have to update zui's PR as well.
Not a big deal at all, I'll just mention this conversation as reference

Reply:
yes, please remove dex, we don't need it, we can just configure it inside oidc.

There are a bunch of tests which should be updated to use 'oidc' instead of 'dex'

Reply:
@babs pls fix the above comment, we can merge your PR after.

Reply:
@babs let us know if you plan to make the change, else we can merge this and patch on top also.

Reply:
I didn't had time to setup a vm dedicated to tests as described in the github action.
Do you have an expected schedule ?

Reply:
I wanted to play all the integration test to make sure I don't break anything but I think I won't have the time for.
I think merge this and patch on top is the wisest option.

Reply:
Please rebase and squash these into a single commit (note you need to sign all your commits after rebase). We need the branch up to date in order for GH to run the tests.

Reply:
rebased and squashed